### PR TITLE
Avoid superfluous RepositoryContent records

### PR DESCRIPTION
**Pull request by gmbnomis**
When adding and removing the same content in a repository version, don't
update the RepositoryContent record (resulting in an superfluous
RepositoryContent for which version_added == version_removed == current
version).  Instead, just delete the existing record.  This fixes two
problems:

1. Such content is shown as added and removed.
2. Re-adding the content again in the same version leads to a
   IntegrityError because the uniqueness constraint is violated.

Handle the "remove then re-add" case analogously.

https://pulp.plan.io/issues/5707
closes #5707

### CODE REVIEW

File: pulpcore/tests/unit/models/test_repository.py [hunk 1]
```diff
@@ -63,3 +63,94 @@ def test_add_and_remove_content(self):
...
+        self.assertCountEqual(added_pks, compress(self.pks, added), added_pks)
+        self.assertCountEqual(removed_pks, compress(self.pks, removed), removed_pks)
+
+    def test_normalize_repository_content(self):
+        """Verify that add_content/delete_content normalize RepositoryContent records properly."""
+        with self.repository.new_version() as version1:
+            version1.add_content(self.content_qs(self.pks[:5]))  # v1 == content 0-4
+
+        # v2 content:
+        # 0 leave as is
```
Comment (daviddavis) [inline pulpcore/tests/unit/models/test_repository.py]: These numbers confused me until I realized that they're indexes. Mind updating this to `# pks[0] ...`?

Comment (daviddavis) [reply pulpcore/tests/unit/models/test_repository.py]: Actually sorry, I see that these are the indexes for each of the assertions below. This test is long and rather hard to follow. Any chance you could break it up into smaller tests?

**Comment by daviddavis** [comment]
This looks great. Thanks for the PR. Should we close out https://github.com/pulp/pulpcore/pull/482 ?

File: pulpcore/tests/unit/models/test_repository.py (hunk 1 above)
Comment (gmbnomis) [reply pulpcore/tests/unit/models/test_repository.py]: Yes, you have a point there. This test was for solution 2/2a originally and I wanted to ensure that all add/remove "histories" are handled correctly by normalization. This is not necessary in here, since there is no need for normalization when saving the version.

I think I will rework this into:

1. Simple test: add, then remove
2. Simple test: remove, then add
3. More complex test with content that is not touched at all, content that is just added or removed and content that is added/removed repeatedly. (to verify that there are no interaction effects and to verify that adding/removing multiple content units with differing "normalization needs" at once works correctly.)

3 will probably be still a little complex, but I will add more documentation.

wdyt?

Comment (daviddavis) [reply pulpcore/tests/unit/models/test_repository.py]: That sounds great. :+1: from me.
